### DISCUSSION TITLE
Confidence interval choice in Kaplan Meier

### DISCUSSION THREAD

**Discussion by claudiocc1** [CamDavidsonPilon/lifelines#1302] [Ideas]
Would it be possible to add a switch in the KM fitter to allow for returning the confidence intervals according to equation 1.2a and 1.2b of 
https://www.math.wustl.edu/~sawyer/handouts/greenwood.pdf
instead of using the exponential formula?

Actually, to be honest, what I am after is just the expression in equation 1.2b (maybe you have a better suggestion on how to easily extract it?).
The reason is that this quantity is necessary to construct the Greenwood-Nam-D'Agostino statistics, see Section 4 of
 https://onlinelibrary.wiley.com/doi/full/10.1002/sim.6428

Thank you so much.
Claudio

**Comment by CamDavidsonPilon** [comment]
Hi @claudiocc1, all the information is available on the `KaplanMeierFitter` object: `kmf.survival_function_` gives you the SF at times (and `kmf.predict` for interpolating, if needed), and `kmf.event_table` gives you the counts. Actually, the `kmf._cumulative_sq_` is the second part of that term exactly.
